test/examples/test_discovery_server.py

```diff
@@ -0,0 +1,54 @@
+import subprocess
+import pytest
+
+"""
+    Expected output test cases for Discovery Server example.
+    Each element is a ternary (pub_args, sub_args, server_args)
+"""
+discovery_server_test_cases = [
+    ('--samples 10', '--samples 10', '--timeout 5', '--id 2 --listening-port 11400 --timeout 1'),
+    ('--connection-port 11500 --samples 10', '--connection-port 11500 --samples 10', '--listening-port 11500 --timeout 5', '--id 2 --listening-port 11400 --timeout 1'),
+    ('--transport tcpv4 --samples 10', '--transport tcpv4 --samples 10', '--transport tcpv4 --timeout 5', '--id 2 --listening-port 11400 --timeout 1'),
+    ('--connection-ds-id 1 --samples 10', '--connection-ds-id 1 --samples 10', '--id 1 --timeout 5', '--id 2 --listening-port 11400 --timeout 1'),
```

Comment:
It would be nice, if instead of this test (which tests the connection-id that is tested also in the following case), we could have a test like the following (that I think is the most complete) for each transport

Reply:
I added the suggestion in [7f9022c](https://github.com/eProsima/Fast-DDS/pull/4913/commits/7f9022cd3f8f6ce27f1c4b982c8476206e5f437d) but we may need to disable `v6` transports if we have troubles with ci.